go-controller/pkg/ovn/egressip.go

```diff
@@ -384,16 +382,23 @@ func (e *egressIPMode) getGatewayRouterJoinIP(node string, wantsIPv6 bool) (net.
 	return nil, fmt.Errorf("node does not have a gateway router IP corresponding to the IP version requested")
 }
 
-func (e *egressIPMode) getPodIPs(pod *kapi.Pod) ([]net.IP, error) {
-	_, podIps, err := util.GetPortAddresses(getPodKey(pod), e.ovnNBClient)
-	if err != nil || len(podIps) == 0 {
+func (e *egressIPMode) getPodIPs(pod *kapi.Pod) []net.IP {
+	if len(pod.Status.PodIPs) == 0 && pod.Status.PodIP == "" {
```

Comment:
You don't need to look at `pod.Status.PodIP` (here or below (or anywhere else)). Unless you're in a cluster much older than anything we support, `PodIPs` will always include `PodIP`.

(Oh, maybe some of the unit tests that manually construct pods get this wrong? If so, fix the test case.)